github/github.go

```diff
@@ -235,6 +235,23 @@ func NewClient(httpClient *http.Client) *Client {
 	return c
 }
 
+// NewEnterpriseClient returns a new GitHub API client (same as NewClient),
+// except it allows injection of custom base and upload urls. These urls might
+// be the same, but for flexibility both can be set separately.
```

Comment:
I think there's an issue here that we need to address. As is, the documentation makes it confusing/unclear as to whether the previous code of creating `NewClient` and setting its `BaseURL`, `UploadURL` fields. I think we should make it clear to users that this is an optional convenience helper, and using it (or changing old code) is not mandatory.

Also, I wouldn't use the word "injection" here, that sounds odd, how about "returns a new GitHub API client with the provided base URL and upload URL"?

The documentation I wrote in https://github.com/google/go-github/issues/756#issuecomment-338877501 did not have these issues IMO, what do you think about using it?

I can also incorporate the "These urls might be the same, but for flexibility both can be set separately" message to it as well, making it:

```Go
// NewEnterpriseClient returns a new GitHub API client with provided
// base URL and upload URL (often the same URL).
// If either URL does not have a trailing slash, one is added automatically.
// If a nil httpClient is provided, http.DefaultClient will be used.
//
// Note that NewEnterpriseClient is a convenience helper only;
// its behavior is equivalent to using NewClient, followed by setting
// the BaseURL and UploadURL fields.
```

Reply:
I agree, this documentation is clearer. I wasn't sure how much detail to go into about what the method did (should have looked back at your suggestion more closely!) I prefer what you have written here, and like the idea of adding in the part about the equivalence to setting the fields directly.